# Predicting Clinical Trial Terminations

**Author: Clement Chan**

---
Notes on this notebook: (Add cool description here!)

- ctg.csv notebook was aquired from clinicaltrials.gov on Feb 16th 2024

### Data Dictionary (Rough outline) But based on clinicaltrials.gov

---

| Column | Description                                  |Data Type|
|-------|--------------------------------------------|-------|
| NCT Number | Unique ID                            | object |
| Study Title | Title of the Clinical Trial           | object |
| Study URL | URL link to the study on clinicaltrials.gov  | object |
| Acronym | An abbreviation used to identify the clinical study | object|
| Study Status | Categorical column displaying the current position of the study | object (could classify into dummies) |
| Brief Summary | Short description of the clinical study (Includes study hypothesis) | object |
| Study Results | Not sure what the result is (but is in yes or no format) | object (turn to binary?)|
| Conditions | Primary Disease or Condition being studied     | object |
| Interventions | The method of the trial???                  | object |
| Primary Outcome Measures | Description of specific primary outcome | object |
| Secondary Outcome Measures | Description of specific secondary outcome | object |
| Sponsor | The corporation or agency that initiates the study | object |
| Collaborators | Other organizations that provide support | object |
| Sex | All: No limit on eligibility based on sex, Male: Only male participants, Female: Only female participants | object (Could change into binary or dummies) |
| Age | Relative age of the participants? Probably need to clarify/change up the name of the values | object (could classify this into dummies)|
| Phases | Clinical trial phase of the study (drug)... need to classify this futher. | object (dummies?)|
| Enrollment | Total estimated number of participants in a study or total actual enrollment? | float |
| Funder Type | Funding type in category? | object (could classify into dummies) |
| Study Type | Interventional, Observational, or Expanded Access | object (definitely dummies) |
| Study Design | Study design based on study type | object |
| Other IDs | Literally other IDs... don't know why this column is useful when we already have NCT number | object |
| Start Date | Estimated/Actual date the first participant was enrolled | object -> date |
| Primary Completion Date | Date the final participant was examined for the first outcome | object -> date |
| Completion Date | Date the final participant was examined | object -> date |
| First Posted |                                        |          |
| Last Update Posted |                                 |         |
| Locations | Where the clinical study will be held | object |


**Importing Libraries**

In [1]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

## Table of Contents
---
1. [Data Wrangling](#wrangle)

>1.1 [First Look At The Dataset](#first)

>1.2 [Cleaning Missing Data](#miss)

2. [Exploratory Data Analysis (EDA)](#EDA)
3.

<a id = 'wrangle'><a/>
## 1. Data Wrangling

---

<a id = 'first'><a/>
### 1.1 First look At The Clinical_Trials Dataset

In [2]:
# Reading the clinical trials dataset
df = pd.read_csv('ctg.csv')

# First 5 rows of dataset
df.head()

NCT Number                                        Study Title  \
0  NCT03630471  Effectiveness of a Problem-solving Interventio...   
1  NCT05992571  Oral Ketone Monoester Supplementation and Rest...   
2  NCT01854671  Investigating the Effect of a Prenatal Family ...   
3  NCT03869671  Pre-exposure Prophylaxis (PrEP) for People Who...   
4  NCT02945371  Tailored Inhibitory Control Training to Revers...   

                                      Study URL Acronym Study Status  \
0  https://clinicaltrials.gov/study/NCT03630471   PRIDE    COMPLETED   
1  https://clinicaltrials.gov/study/NCT05992571     NaN   RECRUITING   
2  https://clinicaltrials.gov/study/NCT01854671     NaN    COMPLETED   
3  https://clinicaltrials.gov/study/NCT03869671     NaN    WITHDRAWN   
4  https://clinicaltrials.gov/study/NCT02945371     REV    COMPLETED   

                                       Brief Summary Study Results  \
0  We will conduct a two-arm individually randomi...            NO   
1  People who report subjective memory complaints...            NO   
2  The purpose of this study is to measure the di...            NO   
3  People who inject drugs (PWID) experience high...            NO   
4  Insufficient inhibitory control is one pathway...            NO   

                                          Conditions  \
0  Mental Health Issue (E.G., Depression, Psychos...   
1                 Cerebrovascular Function|Cognition   
2  Focus: Contraceptive Counseling|Focus: Postpar...   
3                             Intravenous Drug Abuse   
4  Smoking|Alcohol Drinking|Prescription Drug Abu...   

                                       Interventions  \
0  BEHAVIORAL: PRIDE 'Step 1' problem-solving int...   
1           OTHER: Placebo|DIETARY_SUPPLEMENT: β-OHB   
2  OTHER: family planning counseling let by commu...   
3  BEHAVIORAL: PrEP uptake/adherence intervention...   
4  BEHAVIORAL: Person-centered inhibitory control...   

                            Primary Outcome Measures  ...  \
0  Mental health symptoms, The Strengths and Diff...  ...   
1  Brain network connectivity, Functional connect...  ...   
2  Self-reported contraceptive use, 6 months post...  ...   
3  PrEP uptake by self-report, measured using 1 i...  ...   
4  Inhibitory control performance, Task 1, Perfor...  ...   

                                        Study Design  \
0  Allocation: RANDOMIZED|Intervention Model: PAR...   
1  Allocation: RANDOMIZED|Intervention Model: CRO...   
2  Allocation: NON_RANDOMIZED|Intervention Model:...   
3  Allocation: RANDOMIZED|Intervention Model: PAR...   
4  Allocation: RANDOMIZED|Intervention Model: PAR...   

                  Other IDs  Start Date Primary Completion Date  \
0              SANPRIDE_002  2018-08-20              2019-01-20   
1                    rs-KME  2023-10-25                 2024-08   
2                SFPRF13-10     2013-08                 2014-12   
3  H-34960|K01DA043412-01A1     2021-03                 2022-03   
4                 EPCS20613     2014-09                 2016-04   

  Completion Date First Posted Results First Posted  Last Update Posted  \
0      2019-02-28   2018-08-14                  NaN          2019-05-21   
1         2024-08   2023-08-15                  NaN          2023-12-01   
2         2014-12   2013-05-15                  NaN          2015-08-17   
3         2022-03   2019-03-11                  NaN          2021-03-10   
4         2016-05   2016-10-26                  NaN          2016-10-26   

                                           Locations  \
0           Sangath, New Delhi, Delhi, 110016, India   
1  McMaster University, Hamilton, Ontario, L8S 4K...   
2  Palestinian Ministry of Health Maternal Child ...   
3                                                NaN   
4  University of Oregon, Social and Affective Neu...   

                                     Study Documents  
0  Statistical Analysis Plan, https://storage.goo...  
1                                                NaN  
2

In [3]:
# Let's look at the shape of the dataset
f'There are {df.shape[0]} rows and {df.shape[1]} columns.'

'There are 483238 rows and 30 columns.'

In [4]:
# Checking for duplicated data in rows
f'There are {df.duplicated().sum()} duplicated rows.'

'There are 0 duplicated rows.'

In [5]:
# Reading the dataset information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 483238 entries, 0 to 483237
Data columns (total 30 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   NCT Number                  483238 non-null  object 
 1   Study Title                 483238 non-null  object 
 2   Study URL                   483238 non-null  object 
 3   Acronym                     133174 non-null  object 
 4   Study Status                483238 non-null  object 
 5   Brief Summary               482350 non-null  object 
 6   Study Results               483238 non-null  object 
 7   Conditions                  482317 non-null  object 
 8   Interventions               435308 non-null  object 
 9   Primary Outcome Measures    465764 non-null  object 
 10  Secondary Outcome Measures  350835 non-null  object 
 11  Other Outcome Measures      38422 non-null   object 
 12  Sponsor                     483238 non-null  object 
 13  Collaborators 

**Important Notes to Consider:**

- There are 483,238 total rows, but there are large amounts of **missing data** in most of the columns. 
- Most of the columns are objects. (We could change some of these to numerical columns or dates)
- We have 0 duplicated rows, so we don't have to worry about this.

<a id = 'miss'><a/>
### 1.2 Cleaning Missing Data
Let's explore each column ordered by most missing values.

In [6]:
# Finding the sum of NaN values in all columns sorted.
df.isna().sum().sort_values(ascending = False)

Study Documents               450469
Other Outcome Measures        444816
Results First Posted          420886
Acronym                       350064
Collaborators                 325376
Phases                        289380
Secondary Outcome Measures    132403
Locations                      51046
Interventions                  47930
Primary Completion Date        21842
Primary Outcome Measures       17474
Completion Date                16860
Enrollment                      6918
Start Date                      5143
Study Design                    1816
Sex                             1355
Conditions                       921
Study Type                       888
Age                              888
Funder Type                      888
Brief Summary                    888
Other IDs                         57
Study Title                        0
Sponsor                            0
Study Results                      0
First Posted                       0
Study Status                       0
L

In [7]:
# Sorting the missing data by percentages.
((df.isna().sum() / df.shape[0])*100).sort_values(ascending = False)

Study Documents               93.218869
Other Outcome Measures        92.049052
Results First Posted          87.097041
Acronym                       72.441323
Collaborators                 67.332453
Phases                        59.883536
Secondary Outcome Measures    27.399128
Locations                     10.563325
Interventions                  9.918508
Primary Completion Date        4.519926
Primary Outcome Measures       3.616024
Completion Date                3.488964
Enrollment                     1.431593
Start Date                     1.064279
Study Design                   0.375798
Sex                            0.280400
Conditions                     0.190589
Study Type                     0.183760
Age                            0.183760
Funder Type                    0.183760
Brief Summary                  0.183760
Other IDs                      0.011795
Study Title                    0.000000
Sponsor                        0.000000
Study Results                  0.000000


In [3]:
df['Collaborators'].value_counts()

Collaborators
National Cancer Institute (NCI)                                                                                                                                                                                         7920
National Heart, Lung, and Blood Institute (NHLBI)                                                                                                                                                                       1689
National Institute of Mental Health (NIMH)                                                                                                                                                                              1590
National Institute on Drug Abuse (NIDA)                                                                                                                                                                                 1394
Merck Sharp & Dohme LLC                                                                               

Wow that's a lot of missing values. Let's start with the `Study Documents` column.

In [8]:
# Finding out what the values are in the column:
df['Study Documents'].value_counts()

Study Documents
Statistical Analysis Plan, https://storage.googleapis.com/ctgov2-large-docs/71/NCT03630471/SAP_000.pdf                                                                                                                                                                                                                                                                                                      1
Study Protocol and Statistical Analysis Plan, https://storage.googleapis.com/ctgov2-large-docs/95/NCT03347695/Prot_SAP_000.pdf|Informed Consent Form: Informed Consent Form for phase1, https://storage.googleapis.com/ctgov2-large-docs/95/NCT03347695/ICF_001.pdf|Informed Consent Form: Informed Consent Form for phase2, https://storage.googleapis.com/ctgov2-large-docs/95/NCT03347695/ICF_002.pdf    1
Study Protocol, https://storage.googleapis.com/ctgov2-large-docs/83/NCT03850483/Prot_000.pdf|Statistical Analysis Plan, https://storage.googleapis.com/ctgov2-large-docs/83/NCT03850483/SAP_

The `Study Documents` column consists of the clinical design report of each study. However, the column has **450,469** missing values out of 483,238 rows which is about 93% of the dataset N/A, and this column's information would not be significant for our predictive model unless we will be looking into each PDF file, which is not the scope of this project.

Therefore, it is safe to drop this column for our dataset.

In [9]:
# Dropping the `Study Documents` column
df.drop(columns = 'Study Documents', inplace = True)

In [10]:
# Sanity check to see if the column has been dropped
df.columns

Index(['NCT Number', 'Study Title', 'Study URL', 'Acronym', 'Study Status',
       'Brief Summary', 'Study Results', 'Conditions', 'Interventions',
       'Primary Outcome Measures', 'Secondary Outcome Measures',
       'Other Outcome Measures', 'Sponsor', 'Collaborators', 'Sex', 'Age',
       'Phases', 'Enrollment', 'Funder Type', 'Study Type', 'Study Design',
       'Other IDs', 'Start Date', 'Primary Completion Date', 'Completion Date',
       'First Posted', 'Results First Posted', 'Last Update Posted',
       'Locations'],
      dtype='object')

Next, we will look in the `Other Outcome Measures` column since it has **444,816** values.

In [11]:
# Checking out the `Other Outcome Measures` column
df['Other Outcome Measures'].value_counts()

Other Outcome Measures
The dose of CSC vaccine, up to 3 months                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          

The `Other Outcome Measures` column consists of other measures used to evaluate the clinical study. Since we have most of the values missing (444,816) and we have more data evaluating measures in the `Primary Outcome Measures` and `Secondary Outcome Measures` columns, we can safely drop this column as it has too many missing values.

In [12]:
# Dropping the `Study Documents` column
df.drop(columns = 'Other Outcome Measures', inplace = True)

In [13]:
# Sanity check to see if the column has been dropped
df.columns

Index(['NCT Number', 'Study Title', 'Study URL', 'Acronym', 'Study Status',
       'Brief Summary', 'Study Results', 'Conditions', 'Interventions',
       'Primary Outcome Measures', 'Secondary Outcome Measures', 'Sponsor',
       'Collaborators', 'Sex', 'Age', 'Phases', 'Enrollment', 'Funder Type',
       'Study Type', 'Study Design', 'Other IDs', 'Start Date',
       'Primary Completion Date', 'Completion Date', 'First Posted',
       'Results First Posted', 'Last Update Posted', 'Locations'],
      dtype='object')

Next, let's explore the `Results First Posted` column which has **420886** missing values.

In [14]:
# Checking out the `Other Outcome Measures` column
df['Results First Posted'].value_counts()

Results First Posted
2017-04-17    65
2014-05-16    56
2017-04-14    54
2017-09-27    52
2017-04-18    51
              ..
2024-01-24     1
2020-01-01     1
2023-10-28     1
2014-02-23     1
2015-05-16     1
Name: count, Length: 4937, dtype: int64

The `Results First Posted` column is comprised of dates where the clinical study first posts their results to the public. This information is not really related to our goal of predicting trial terminations, and since most of the data is missing, we can also drop this column.

In [15]:
# Dropping the `Results First Posted` column
df.drop(columns = 'Results First Posted', inplace = True)

In [16]:
# Sanity check to see if the column has been dropped
df.columns

Index(['NCT Number', 'Study Title', 'Study URL', 'Acronym', 'Study Status',
       'Brief Summary', 'Study Results', 'Conditions', 'Interventions',
       'Primary Outcome Measures', 'Secondary Outcome Measures', 'Sponsor',
       'Collaborators', 'Sex', 'Age', 'Phases', 'Enrollment', 'Funder Type',
       'Study Type', 'Study Design', 'Other IDs', 'Start Date',
       'Primary Completion Date', 'Completion Date', 'First Posted',
       'Last Update Posted', 'Locations'],
      dtype='object')

Next, let's examine the `Acronym` column which has **350,064** missing values.

In [17]:
# Checking the values in `Acronym` column
df['Acronym'].value_counts()

Acronym
IMPACT       129
COVID-19     122
SMART        111
RCT           88
STAR          78
            ... 
boron_gel      1
CUMACA-M       1
PR11           1
NutriCim       1
AFOCUFF        1
Name: count, Length: 104724, dtype: int64

In [18]:
# We can calculate the % of missing data in the Acronym column by dividing by the total number of rows
(df['Acronym'].isna().sum() / df.shape[0])*100

72.4413229092083

Since the `Acronym` column has more than 70% of data missing and the acronym is not very useful for our models, we can also drop this column from our dataset.

In [19]:
# Dropping the `Acronym` column
df.drop(columns = 'Acronym', inplace = True)

In [20]:
# Sanity check to see if the column has been dropped
df.columns

Index(['NCT Number', 'Study Title', 'Study URL', 'Study Status',
       'Brief Summary', 'Study Results', 'Conditions', 'Interventions',
       'Primary Outcome Measures', 'Secondary Outcome Measures', 'Sponsor',
       'Collaborators', 'Sex', 'Age', 'Phases', 'Enrollment', 'Funder Type',
       'Study Type', 'Study Design', 'Other IDs', 'Start Date',
       'Primary Completion Date', 'Completion Date', 'First Posted',
       'Last Update Posted', 'Locations'],
      dtype='object')

Great! Now we can check out the missing values from the `Collaborators` column which has **325,376** values missing.

In [21]:
df['Collaborators'].isna().sum()

325376

In [22]:
(df['Collaborators'].isna().sum() / df.shape[0])*100

67.33245315972668

In [23]:
# Checking the values in `Collaborators` column
df['Collaborators'].value_counts()

Collaborators
National Cancer Institute (NCI)                                                                                                                                                                                         7920
National Heart, Lung, and Blood Institute (NHLBI)                                                                                                                                                                       1689
National Institute of Mental Health (NIMH)                                                                                                                                                                              1590
National Institute on Drug Abuse (NIDA)                                                                                                                                                                                 1394
Merck Sharp & Dohme LLC                                                                               

Could potentially use this column! Change missing values into "No collaborators? and maybe group the collaborators with 1 value into others?

Next, Let's looking at the missing data in the `Phases` column.

In [24]:
f"There is {(df['Phases'].isna().sum() / df.shape[0])*100}% of missing data in the `Phases` column."

'There is 59.88353564910044% of missing data in the `Phases` column.'

In [25]:
# Checking the values in `Collaborators` column
df['Phases'].value_counts()

Phases
PHASE2           56500
PHASE1           41845
PHASE3           37654
PHASE4           31876
PHASE1|PHASE2    14385
PHASE2|PHASE3     6697
EARLY_PHASE1      4901
Name: count, dtype: int64

From the clinicaltrials.gov documentation, Trials without phases are listed as N/A which means that we can change the missing values into a string 'Trial Without Phases'.

In [26]:
# We can use the .fillna() function to fill the NaN values.
# df['Phases'] = df['Phases'].fillna('Trial Without Phases')

In [27]:
pd.options.display.max_columns = None

In [28]:
df.head()

NCT Number                                        Study Title  \
0  NCT03630471  Effectiveness of a Problem-solving Interventio...   
1  NCT05992571  Oral Ketone Monoester Supplementation and Rest...   
2  NCT01854671  Investigating the Effect of a Prenatal Family ...   
3  NCT03869671  Pre-exposure Prophylaxis (PrEP) for People Who...   
4  NCT02945371  Tailored Inhibitory Control Training to Revers...   

                                      Study URL Study Status  \
0  https://clinicaltrials.gov/study/NCT03630471    COMPLETED   
1  https://clinicaltrials.gov/study/NCT05992571   RECRUITING   
2  https://clinicaltrials.gov/study/NCT01854671    COMPLETED   
3  https://clinicaltrials.gov/study/NCT03869671    WITHDRAWN   
4  https://clinicaltrials.gov/study/NCT02945371    COMPLETED   

                                       Brief Summary Study Results  \
0  We will conduct a two-arm individually randomi...            NO   
1  People who report subjective memory complaints...            NO   
2  The purpose of this study is to measure the di...            NO   
3  People who inject drugs (PWID) experience high...            NO   
4  Insufficient inhibitory control is one pathway...            NO   

                                          Conditions  \
0  Mental Health Issue (E.G., Depression, Psychos...   
1                 Cerebrovascular Function|Cognition   
2  Focus: Contraceptive Counseling|Focus: Postpar...   
3                             Intravenous Drug Abuse   
4  Smoking|Alcohol Drinking|Prescription Drug Abu...   

                                       Interventions  \
0  BEHAVIORAL: PRIDE 'Step 1' problem-solving int...   
1           OTHER: Placebo|DIETARY_SUPPLEMENT: β-OHB   
2  OTHER: family planning counseling let by commu...   
3  BEHAVIORAL: PrEP uptake/adherence intervention...   
4  BEHAVIORAL: Person-centered inhibitory control...   

                            Primary Outcome Measures  \
0  Mental health symptoms, The Strengths and Diff...   
1  Brain network connectivity, Functional connect...   
2  Self-reported contraceptive use, 6 months post...   
3  PrEP uptake by self-report, measured using 1 i...   
4  Inhibitory control performance, Task 1, Perfor...   

                          Secondary Outcome Measures  \
0  Mental health symptoms, The adolescent-reporte...   
1  Working memory, Computer battery to assess wor...   
2  Intent to use contraception in the future, 6 m...   
3  Participant satisfaction with intervention con...   
4  Far transfer to a task related to inhibitory c...   

                                      Sponsor  \
0                                     Sangath   
1                         McMaster University   
2  Planned Parenthood League of Massachusetts   
3                           Boston University   
4                        University of Oregon   

                                       Collaborators     Sex  \
0  Harvard Medical School (HMS and HSDM)|London S...     ALL   
1  Alzheimer's Society of Brant, Haldimand Norfol...     ALL   
2          Society for Family Planning Research Fund  FEMALE   
3            National Institute on Drug Abuse (NIDA)     ALL   
4                                                NaN     ALL   

                         Age Phases  Enrollment Funder Type      Study Type  \
0               CHILD, ADULT    NaN       250.0       OTHER  INTERVENTIONAL   
1         ADULT, OLDER_ADULT    NaN        30.0       OTHER  INTERVENTIONAL   
2  CHILD, ADULT, OLDER_ADULT    NaN       176.0       OTHER  INTERVENTIONAL   
3         ADULT, OLDER_ADULT    NaN         0.0       OTHER  INTERVENTIONAL   
4                      ADULT    NaN       103.0       OTHER  INTERVENTIONAL   

                                        Study Design  \
0  Allocation: RANDOMIZED|Intervention Model: PAR...   
1  Allocation: RANDOMIZED|Intervention Model: CRO...   
2  Allocation: NON_RANDOMIZED|Intervention Model:...   
3  Allocation: RANDOMIZED|Intervention Model: PAR...   


In [29]:
# Finding the number of NaN (missing values)
f"There are {df['Brief Summary'].isna().sum()} missing values in the `Brief Summary` column."

'There are 888 missing values in the `Brief Summary` column.'

In [30]:
# diving into the values of `Brief Summary`
df['Brief Summary'].value_counts()

Brief Summary
To evaluate the Sun Protection Factor efficacy on human skin.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

There are only **888** missing values in this column, and it will be usefull to use NLP keyword matching to match specific words together which will increase the predictivity of our model.

Next, let's dive into the `Conditions` column.

In [31]:
# Finding the number of NaN (missing values)
f"There are {df['Conditions'].isna().sum()} missing values in the `Conditions` column."

'There are 921 missing values in the `Conditions` column.'

In [32]:
# Looking into the values of `Conditions`
df['Conditions'].value_counts()

Conditions
Healthy                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           7996
Breast Cancer                                                         

There are **921** missing values in the `Conditions` column. This column will also be very useful in grouping certain conditions or diseases that are being studied primarily.

Next, we will look at the `Interventions` column.

In [33]:
# Finding the number of NaN (missing values)
f"There are {df['Interventions'].isna().sum()} missing values in the `Interventions` column."

'There are 47930 missing values in the `Interventions` column.'

In [34]:
# Looking into the values of `Conditions`
df['Interventions'].value_counts()

Interventions
OTHER: No intervention                                                                          1083
OTHER: no intervention                                                                           410
OTHER: Questionnaire                                                                             318
OTHER: Exercise                                                                                  273
OTHER: No Intervention                                                                           249
                                                                                                ... 
DEVICE: Patient specific instrumentation (MRI)|DEVICE: Patient specific instrumentation (CT)       1
DRUG: Bone Morphogenetic Protein 2|PROCEDURE: Autologous bone graft                                1
OTHER: HaRTS-TRENDS|OTHER: Standard Care (SC)                                                      1
DIAGNOSTIC_TEST: Stroke simulation and machine learning                      

There are **47,930** missing values in the `Intervention` column. Don't know what this is yet... will figure out this later.

Next is `Primary Outcome Measures` column:

In [35]:
# Finding the number of NaN (missing values)
f"There are {df['Primary Outcome Measures'].isna().sum()} missing values in the `Primary Outcome Measures` column."

'There are 17474 missing values in the `Primary Outcome Measures` column.'

In [36]:
# Looking into the column
df['Primary Outcome Measures'].value_counts()

Primary Outcome Measures
Bioequivalence, within 30 days                                                                                                                                                                                                                   118
Bioequivalence                                                                                                                                                                                                                                    68
Overall survival                                                                                                                                                                                                                                  49
Minimal Erythema Dose (MED), Up to 15 minutes|Minimal Persistent Pigment Darkening Dose (MPPD), Up to 15 minutes                                                                                                                                

There are **17,474** missing values in the `Primary Outcome Measures` column. Don't know what this is yet... will figure out this later.

Next is `Secondary Outcome Measures` column:

In [37]:
# Finding the number of NaN (missing values)
f"There are {df['Secondary Outcome Measures'].isna().sum()} missing values in the `Secondary Outcome Measures` column."

'There are 132403 missing values in the `Secondary Outcome Measures` column.'

In [38]:
# Looking into the column
df['Primary Outcome Measures'].value_counts()

Primary Outcome Measures
Bioequivalence, within 30 days                                                                                                                                                                                                                   118
Bioequivalence                                                                                                                                                                                                                                    68
Overall survival                                                                                                                                                                                                                                  49
Minimal Erythema Dose (MED), Up to 15 minutes|Minimal Persistent Pigment Darkening Dose (MPPD), Up to 15 minutes                                                                                                                                

There are **132,403** missing values in the `Secondary Outcome Measures` column. Don't know what this is yet... will figure out this later.

Next is `Other Outcome Measures` column:

There are **444,816** missing values in the `Other Outcome Measures` column. Don't know what this is yet... will figure out this later. Most of the data in this column is missing. So this column is a potential DROP.

Next is `Collaborators` column:

In [39]:
# Finding the number of NaN (missing values)
f"There are {df['Collaborators'].isna().sum()} missing values in the `Collaborators` column."

'There are 325376 missing values in the `Collaborators` column.'

In [40]:
# Looking into the values of the column
df['Collaborators'].value_counts()

Collaborators
National Cancer Institute (NCI)                                                                                                                                                                                         7920
National Heart, Lung, and Blood Institute (NHLBI)                                                                                                                                                                       1689
National Institute of Mental Health (NIMH)                                                                                                                                                                              1590
National Institute on Drug Abuse (NIDA)                                                                                                                                                                                 1394
Merck Sharp & Dohme LLC                                                                               

There are large amounts of missing data, **(325,376)** values of missing data. Will need to analyze this further, but this column is a potential drop, unless the collaborator (NCI) is useful since it's present in 7920 rows of data.

Next, we will look at the `Sex` column (There doesn't seem to be much missing data).

In [41]:
# Finding the number of NaN (missing values)
f"There are {df['Sex'].isna().sum()} missing values in the `Sex` column."

'There are 1355 missing values in the `Sex` column.'

In [42]:
# Finding out what the values are in the column:
df['Sex'].value_counts()

Sex
ALL       414720
FEMALE     46562
MALE       20601
Name: count, dtype: int64

Need to look into this further to decide what to change the NaN values into... But most of the studies allow both male and female sex in the studies.

Next, let's look at the `Age` column.

In [43]:
# Finding the number of NaN (missing values)
f"There are {df['Age'].isna().sum()} missing values in the `Age` column."

'There are 888 missing values in the `Age` column.'

In [44]:
# Finding out what the values are in the column:
df['Age'].value_counts()

Age
ADULT, OLDER_ADULT           315511
ADULT                         64792
CHILD, ADULT, OLDER_ADULT     45657
CHILD                         29103
CHILD, ADULT                  20014
OLDER_ADULT                    7273
Name: count, dtype: int64

There are very few missing values **(888)** missing values, however we will need to clean up the values name as they are confusing...

Next, we will look at the `Phases` column.

In [45]:
# Finding the number of NaN (missing values)
f"There are {df['Phases'].isna().sum()} missing values in the `Phases` column."

'There are 289380 missing values in the `Phases` column.'

In [46]:
# Finding out what the values are in the column:
df['Phases'].value_counts()

Phases
PHASE2           56500
PHASE1           41845
PHASE3           37654
PHASE4           31876
PHASE1|PHASE2    14385
PHASE2|PHASE3     6697
EARLY_PHASE1      4901
Name: count, dtype: int64

Very useful information to the model, could classify whether the study was terminated in different phases. However, there is a lot of missing values of **289,380**. We need to find out how to classify these...

Next, we will look into the `Enrollment` column.

In [47]:
# Finding the number of NaN (missing values)
f"There are {df['Enrollment'].isna().sum()} missing values in the `Enrollment` column."

'There are 6918 missing values in the `Enrollment` column.'

In [48]:
# Finding out what the values are in the column:
df['Enrollment'].value_counts()

Enrollment
30.0        15910
60.0        15792
100.0       14534
40.0        14202
20.0        13928
            ...  
419734.0        1
25600.0         1
254432.0        1
10385.0         1
290286.0        1
Name: count, Length: 8156, dtype: int64

On the site it says total estimation enrollment OR actual number of participants enrolled... Need to find out which case is what. Perhaps make a new column with updated information.

Next, we will explore the `Funder Type` column.

In [49]:
# Finding the number of NaN (missing values)
f"There are {df['Funder Type'].isna().sum()} missing values in the `Funder Type` column."

'There are 888 missing values in the `Funder Type` column.'

In [50]:
# Finding out what the values are in the column:
df['Funder Type'].value_counts()

Funder Type
OTHER        336155
INDUSTRY     113018
OTHER_GOV     12376
NIH           11193
FED            4526
NETWORK        4409
INDIV           575
UNKNOWN          95
AMBIG             3
Name: count, dtype: int64

This also has 888 missing values, could explore why there are columns with 888 missing values. For this specifically, could combine ambig to unknown and missing values into unknown.

Next, let's look into the `Study Type` column.

In [51]:
# Finding the number of NaN (missing values)
f"There are {df['Study Type'].isna().sum()} missing values in the `Study Type` column."

'There are 888 missing values in the `Study Type` column.'

In [52]:
# Finding out what the values are in the column:
df['Study Type'].value_counts()

Study Type
INTERVENTIONAL     371242
OBSERVATIONAL      110180
EXPANDED_ACCESS       928
Name: count, dtype: int64

Also has 888 missing values, not sure where we could put these, but 3 categories is definitely useful for the model. (Can talk about what each does further...)

Next, we will look into the `Study Design` column.

In [53]:
# Finding the number of NaN (missing values)
f"There are {df['Study Design'].isna().sum()} missing values in the `Study Design` column."

'There are 1816 missing values in the `Study Design` column.'

In [54]:
# Finding out what the values are in the column:
df['Study Design'].value_counts()

Study Design
Observational Model: |Time Perspective: p                                                                                                                          110180
Allocation: NA|Intervention Model: SINGLE_GROUP|Masking: NONE|Primary Purpose: TREATMENT                                                                            52601
Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: NONE|Primary Purpose: TREATMENT                                                                        42532
Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: QUADRUPLE (PARTICIPANT, CARE_PROVIDER, INVESTIGATOR, OUTCOMES_ASSESSOR)|Primary Purpose: TREATMENT     21277
Allocation: RANDOMIZED|Intervention Model: PARALLEL|Masking: DOUBLE (PARTICIPANT, INVESTIGATOR)|Primary Purpose: TREATMENT                                          13659
                                                                                                                                         

Need to figure out what to do with these unknowns, but most of these are based on models: Observational or Intervention model...

Next, let's dive into the `Other IDs` column.

In [55]:
# Finding the number of NaN (missing values)
f"There are {df['Other IDs'].isna().sum()} missing values in the `Other IDs` column."

'There are 57 missing values in the `Other IDs` column.'

In [56]:
# Finding out what the values are in the column:
df['Other IDs'].value_counts()

Other IDs
1                               261
001                             208
01                               96
2                                87
002                              77
                               ... 
HBBL-01                           1
3207-003|MK3207-003|2007_599      1
REB17-730                         1
B83-003                           1
SLX53                             1
Name: count, Length: 473060, dtype: int64

Need to figure out how these IDs interact with the NCT number... why do we need so many IDs. Some of these don't make sense, like why are 261 IDs just 1 or 001. We should drop this column (high priority)...

Let's look at the `Start Date` column next.

In [57]:
# Finding the number of NaN (missing values)
f"There are {df['Start Date'].isna().sum()} missing values in the `Start Date` column."

'There are 5143 missing values in the `Start Date` column.'

In [58]:
# Finding out what the values are in the column:
df['Start Date'].value_counts()

Start Date
2014-01       2087
2015-01       2025
2013-01       1959
2012-01       1937
2011-01       1924
              ... 
1998-03-24       1
2000-03-29       1
2009-04-18       1
1999-12-06       1
2012-10-28       1
Name: count, Length: 8659, dtype: int64

Start Date would be very important for time series analysis in the model. Since we can't really fill the NaN values with a number, maybe an unknown start date is best.

Next, let's explore the `Primary Completion Date` column.

In [59]:
# Finding the number of NaN (missing values)
f"There are {df['Primary Completion Date'].isna().sum()} missing values in the `Primary Completion Date` column."

'There are 21842 missing values in the `Primary Completion Date` column.'

In [60]:
# Finding out what the values are in the column:
df['Primary Completion Date'].value_counts()

Primary Completion Date
2023-12-31    3496
2024-12-31    2992
2015-12       2971
2024-12       2575
2014-12       2563
              ... 
2008-09-26       1
2029-05-29       1
2011-09-10       1
2000-06-09       1
2006-06-20       1
Name: count, Length: 9254, dtype: int64

Same as previous column but for completed first outcome dates. Need to compare missing values with completed vs terminated outcomes.

Let's explore the `Completion Date` column next.

In [61]:
# Finding the number of NaN (missing values)
f"There are {df['Completion Date'].isna().sum()} missing values in the `Completion Date` column."

'There are 16860 missing values in the `Completion Date` column.'

In [62]:
# Finding out what the values are in the column:
df['Completion Date'].value_counts()

Completion Date
2024-12-31    3825
2023-12-31    3470
2024-12       3271
2015-12       2898
2014-12       2687
              ... 
2013-06-13       1
2030-02-22       1
2034-03-30       1
1995-08-07       1
2004-11-29       1
Name: count, Length: 9781, dtype: int64

There's less missing data here than the `Primary Completion Date`. I wonder if there's a reason why. Let's look at this more later.

**In terms of the First Posted, Results Posted, and Last Update Posted** are not that useful for us in this model and there are a lot of missing data in results first posted... So we will probably remove all 3 of these columns.

Next, we will look at the `Locations` column.

In [63]:
# Finding the number of NaN (missing values)
f"There are {df['Locations'].isna().sum()} missing values in the `Locations` column."

'There are 51046 missing values in the `Locations` column.'

In [64]:
# Finding out what the values are in the column:
df['Locations'].value_counts()

Locations
National Institutes of Health Clinical Center, 9000 Rockville Pike, Bethesda, Maryland, 20892, United States                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  

I think the simplest way to fill the NaN values is to create an unknown value for the locations.

Final column to look at is `Study Documents`